### PR TITLE
Single include search path

### PR DESCRIPTION
**Pull request by xpol**
This PR makes all sources (c, cpp, h, hpp, inl) in cocos directory include cocos2d-x headers in with path relative to cocos directory (As discussed in #15250 the No. 1 issue)
1. Add a include linter script in `tools/coding-style/include-linter.py`
2. Fixes 1605 include path in 541 files using above script.

~~TODO: May cocos2d-x team add `tools/coding-style/include-linter.py` in CI script.~~

### CODE REVIEW

File: tools/coding-style/include-linter.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
```
Comment (zilongshanren) [inline tools/coding-style/include-linter.py:1]: Great job @xpol , I do like this idea to lint the include files. @ricardoquesada  Could you please have a look at this PR? thanks.

**Comment by xpol** [comment]
I have locally added running of `tools/coding-style/include-linter.py` in `tools/jenkins-scripts/do-pull-request-builder.py`. If you like, I can include it in this PR or make another.

**Comment by ricardoquesada** [comment]
good for me. Once the build is green, we can merge it.

**Comment by xpol** [comment]
BTW, I think `cocos/platform/winrt/pch.h` and `cocos/platform/winrt/pch.h` should be both included in `cocos/platform/win8.1-universal/pch.cpp` with some macro condition.

Any macro that I can use to do this?

**Comment by xpol** [comment]
OK, build passed.
It's time to enable the linter check in CI script.

**Comment by zilongshanren** [comment]
@xpol  Good :+1:  After the CI is green, I will merge this PR, big thanks for the linter script.

**Comment by zilongshanren** [comment]
@xpol  The next step is to clean up the search paths in various project, including the templates.
Eg. Xcode templates, VS templates, Android.mk, cmake file etc...

**Comment by xpol** [comment]
@zilongshanren 
Yes, the search path can be cleaned now. Wait my next PR.

**Comment by zilongshanren** [comment]
@xpol  few search paths could also boost compile time :smile:

**Comment by zilongshanren** [comment]
@xpol 
There are some bugs in your linter file:

```
platform\win8.1-universal\pch.cpp:
    #include "pch.h" should be #include "platform\win8.1-universal\pch.h"
platform\winrt\pch.cpp:
    #include "pch.h" should be #include "platform\winrt\pch.h"
Total: 2 errors in 2 files
Rerun this script with -f to fixes these errors
prepare to downloading ...
return value is 1
```

**Comment by zilongshanren** [comment]
[ci rebuild]

**Comment by xpol** [comment]
The build of cocos2d-x takes too much time.
One reason is #15250 see [GotW #7](http://www.gotw.ca/gotw/007.htm).
_Precompiled headers is just workaround to fix this issue, but it will introduce new issue (forget to include necessary header)_.

**Pull request merged**